prime-router/src/main/kotlin/transport/RESTTransport.kt

```diff
@@ -338,20 +327,31 @@ class RESTTransport(private val httpClient: HttpClient? = null) : ITransport {
      * @param httpClient the HTTP client to make the call
      */
     suspend fun getAuthTokenWithUserApiKey(
-        restUrl: String,
+        restTransportInfo: RESTTransportType,
         credential: UserApiKeyCredential,
         logger: Logger,
         httpClient: HttpClient
     ): TokenInfo {
         httpClient.use { client ->
             val tokenInfo: TokenInfo = client.submitForm(
-                restUrl,
+                restTransportInfo.authTokenUrl,
                 formParameters = Parameters.build {
-                    append("grant_type", "client_credentials")
-                    append("client_id", credential.user)
-                    append("client_secret", credential.apiKey)
+                    if (restTransportInfo.parameters!!.isEmpty()) {
```

Comment:
If you change `parameters` to be non-nullable, then you can remove the `!!`
```suggestion
                    if (restTransportInfo.parameters.isEmpty()) {
```
If you don't make `parameters` non-nullable, then maybe something like `restTransportInfo.parameters?.isEmpty() != false` would be safer.

Reply:
As I explained above, parameters can be null.

Reply:
If `parameters` can be `null`, this check will fail due a null pointer exception. `!!` asserts that the variable is _not_ null, and will throw an exception if that is untrue. Based on the other conversation, since the default for `parameters` is `emptyMap()` I don't think this will actually be `null`, but if you think it might, then we should still use `restTransportInfo.parameters?.isEmpty() != false` to prevent execptions.

Reply:
Yes, you are correct.  As I mentioned above, I used null wrongly.

Reply:
Got it, in that case, we should make the variable non-nullable and remove the `!!` here.